core/src/main/scala/org/http4s/Message.scala

```diff
@@ -66,7 +66,7 @@ sealed trait Message extends MessageOps { self =>
   def withBody[T](b: T)(implicit w: EntityEncoder[T]): Task[Self] = {
     w.toEntity(b).map { entity =>
       val hs = entity.length match {
-        case Some(l) => `Content-Length`(l)::w.headers.toList
+        case Some(l) => `Content-Length`(l).fold(_ => w.headers.toList, _ :: w.headers.toList)
```

Comment:
This is maybe a more general question. I've seen in the codebase log calls in certain places but you could think that they should be wrapped on e.g. `Task` as side effects. I'm ok doing it either way but just want to be consistent